core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinder.java

```diff
@@ -595,11 +596,11 @@ private KojiBuild findBestBuildFromCandidates(List<KojiBuild> candidates, List<K
                     buildCache.put(cacheRpmBuildInfo.getBuildInfo().getId(), cacheRpmBuildInfo);
                 }
             } else {
-                if (cacheManager == null || (cacheArchiveInfos = checksumCaches.get(ChecksumType.md5)
-                        .get(checksum.getValue())) == null) {
+                if (cacheManager == null || checksumCaches.get(ChecksumType.md5).get(checksum.getValue()) == null) {
                     LOGGER.debug("Add checksum {} to list", checksum);
                     checksums.add(entry);
                 } else {
+                    cacheArchiveInfos = checksumCaches.get(ChecksumType.md5).get(checksum.getValue()).getData();
```

Comment:
We should avoid possible double call to `checksumCaches.get(ChecksumType.md5).get(checksum.getValue())` here.

Reply:
Good point!